dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java

```diff
@@ -208,24 +246,25 @@ private <T> void toObjectNodes( List<T> objects, List<FieldPath> fieldPaths, Use
         // In case we get a proxied object in we can't just use o.getClass(), we
         // need to figure out the real class name by using HibernateProxyUtils.
         Object firstObject = objects.iterator().next();
-        fieldPathHelper.apply( fieldPaths, HibernateProxyUtils.getRealClass( firstObject ) );
+        List<FieldPath> expandedFilter = fieldPathHelper.apply( filter,
+            HibernateProxyUtils.getRealClass( firstObject ) );
 
-        SimpleFilterProvider filterProvider = getSimpleFilterProvider( fieldPaths, isSkipSharing );
+        SimpleFilterProvider filterProvider = getSimpleFilterProvider( expandedFilter, isSkipSharing );
```

Comment:
`expandedFilter` is too much focused on the aspect that it has been expanded, the question I ask when a want to use a variable is more "what do we get?" not "what has happened?". I'd call this `includedPaths` or just `paths` or `fieldPaths`.